Mathlib/Order/Bounds/Basic.lean

```diff
@@ -1702,17 +1705,31 @@ The dual notion
 
 does not appear to play a significant role in the literature, so is omitted here.
 -/
-def ScottContinuous (f : α → β) : Prop :=
-  ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → ∀ ⦃a⦄, IsLUB d a → IsLUB (f '' d) (f a)
-#align scott_continuous ScottContinuous
+def DScottContinuous (D : Set (Set α)) (f : α → β) : Prop :=
+  ∀ ⦃d : Set α⦄, d.Nonempty → DirectedOn (· ≤ ·) d → d ∈ D → ∀ ⦃a⦄, IsLUB d a → IsLUB (f '' d) (f a)
+#align scott_continuous DScottContinuous
 
-protected theorem ScottContinuous.monotone (h : ScottContinuous f) : Monotone f := by
+lemma DScottContinuous.LE (D₁ D₂ : Set (Set α)) (hD : D₁ ≤ D₂) {f : α → β}
+    (hf : DScottContinuous D₂ f) : DScottContinuous D₁ f :=
+  fun _ hd₁ hd₂ hdD₁ _ hda => hf hd₁ hd₂ (hD hdD₁) hda
+
+protected theorem DScottContinuous.monotone (D : Set (Set α)) (hD : ∀ a b : α, a ≤ b → {a, b} ∈ D)
+    (h : DScottContinuous D f) : Monotone f := by
   refine' fun a b hab =>
-    (h (insert_nonempty _ _) (directedOn_pair le_refl hab) _).1
+    (h (insert_nonempty _ _) (directedOn_pair le_refl hab) (hD a b hab) _).1
       (mem_image_of_mem _ <| mem_insert _ _)
   rw [IsLUB, upperBounds_insert, upperBounds_singleton,
     inter_eq_self_of_subset_right (Ici_subset_Ici.2 hab)]
   exact isLeast_Ici
-#align scott_continuous.monotone ScottContinuous.monotone
+#align scott_continuous.monotone DScottContinuous.monotone
+
+/-- A function between preorders is said to be Scott continuous if it preserves `IsLUB` on directed
+sets. It can be shown that a function is Scott continuous if and only if it is continuous wrt the
+Scott topology.
+-/
+def ScottContinuous (f : α → β) := DScottContinuous univ f
```

Comment:
I suspect it's better to not change the definition to `DScottContinuous` but instead show `@[simp] lemma dscottContinuous_univ : DScottContinuous univ f \iff ScottContinuous`

Reply:
Done, thanks. That takes 2 files out of the diff.